api/src/metrics/lib/create-metrics.ts

```diff
@@ -0,0 +1,228 @@
+import { useEnv } from '@directus/env';
+import { toArray } from '@directus/utils';
+import { randomUUID } from 'node:crypto';
+import { Readable } from 'node:stream';
+import { AggregatorRegistry, Counter, Histogram } from 'prom-client';
+import { getCache } from '../../cache.js';
+import { hasDatabaseConnection } from '../../database/index.js';
+import { redisConfigAvailable, useRedis } from '../../redis/index.js';
+import { getStorage } from '../../storage/index.js';
+
+export function createMetrics() {
+	const env = useEnv();
+
+	const registry = new AggregatorRegistry();
+
+	async function generate() {
+		const checkId = randomUUID();
+
+		await Promise.all([
+			trackDatabaseMetric(),
+			trackCacheMetric(checkId),
+			trackRedisMetric(checkId),
+			trackStorageMetric(checkId),
+		]);
+	}
+
+	function readAll(): Promise<string> {
+		return registry.metrics();
```

Comment:
For the AggregatorRegistry to work we need to call clusterMetrics() otherwise we only get the local ones.

Reply:
Upon further investigation it is not possible to use `clusterMetrics` in a `pm2` context as it utilizes the cluster module [which doesn't work with pm2](https://github.com/Unitech/pm2/issues/2885#issue-229467279).

Due to this we will need to add our own custom IPC logic as originally proposed

Reply:
IPC logic added in [66979cd](https://github.com/directus/directus/pull/24233/commits/66979cd12d952c3dd0a98c06f1c0347462c12622) and fine tuned a bit in [97afd49](https://github.com/directus/directus/pull/24233/commits/97afd49f926e5c52d495246008d7991cd19e261c).